internal/dot/graph.go

```diff
@@ -252,6 +274,59 @@ func (dg *Graph) addToGroup(r *Result, id CtorID) {
 	group.Results = append(group.Results, r)
 }
 
+// Prune removes elements from the graph that do not have failed results.
+// Removing elements that do not have failing results makes the graph easier to debug, since non-failing
+// nodes and edges can clutter the graph and don't help the user debug.
+func (dg *Graph) Prune() {
```

Comment:
let's rename this to something that makes the behavior slightly more obvious without reading the docs. Something like `PruneSuccess` or `FilterFailed`? (There are likely better names than that too.)

Reply:
I think PruneSuccess is a great name!